### PR TITLE
[openvpn] Go hooks

### PR DESCRIPTION
**Pull request by vitaliy-sn**
## Description

This MR is a part of "Global" Deckhouse refactoring.

## Why do we need it, and what problem does it solve?

It is aimed to improve UX and DX.

Go is more reliable and suitable to describe cloud infrastructure components, especially in distributed systems/environments
Open API specs will help us with bounding a contract between user and deckhouse, and between hooks and helm
An Open API spec for a module will become the single source of truth (documenting, validating, testing)
More tests, more stability

## Changelog entries

```changes
section: openvpn
type: feature
summary: Hooks are rewritten in Go.
```

### CODE REVIEW

File: ee/fe/modules/500-openvpn/hooks/copy_rbd_secret.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+					{
+						Field:    "type",
+						Operator: "Equals",
+						Value:    "kubernetes.io/rbd",
+					},
+				},
+			},
+		},
+	},
+}, copyRBDSecretHandler)
```
Comment (nabokihms) [inline ee/fe/modules/500-openvpn/hooks/copy_rbd_secret.go:85]: There is the same hook in the `prometheus` module. It is required to move it to the go_lib then.

File: ee/fe/modules/500-openvpn/hooks/generate_password.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	"github.com/deckhouse/deckhouse/go_lib/pwgen"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue:        "/modules/openvpn/generate_password",
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 10},
+}, generatePassword)
+
+func generatePassword(input *go_hook.HookInput) error {
```
Comment (nabokihms) [inline ee/fe/modules/500-openvpn/hooks/generate_password.go:22]: Let's also move this hook to the shared library (and replace it for other modules).

File: ee/fe/modules/500-openvpn/hooks/set_push_routes.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	serviceSubnet := input.Values.Get("global.discovery.serviceSubnet").String()
+
+	routeList = append(routeList, podSubnet)
+	routeList = append(routeList, serviceSubnet)
+
+	if !input.ConfigValues.Exists("openvpn.pushToClientRoutes") {
+		input.ConfigValues.Set("openvpn.pushToClientRoutes", routeList)
+	}
+
+	return nil
```
Comment (nabokihms) [inline ee/fe/modules/500-openvpn/hooks/set_push_routes.go:30]: What with the `KUBE_NODES_INFO=$(kubectl get nodes -o json)`? Is it crucial to have routes to nodes?

**Review by nabokihms — CHANGES_REQUESTED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

**Review by nabokihms — CHANGES_REQUESTED** [review]
@vitaliy-sn, could you please rebase your branch onto the main branch?

Integration tests fail with the following error
```
{"event.id":"220b1640-7d0e-4932-85f4-d1f13fd5c0ce","event.triggered-by.binding":"kubernetes","event.triggered-by.binding.name":"ready_not_master_nodes","event.triggered-by.hook":"cluster_is_bootstrapped.go","event.triggered-by.watchEvent":"Modified","level":"error","module":"helm","msg":"ModuleRun failed in phase 'Startup'. Requeue task to retry after delay. Failed count is 36. Error: module hook '013-helm/hooks/startup_metrics.go' failed: Get \"https://prometheus.d8-monitoring:9090/api/v1/query?query=resource_versions_compatibility\": dial tcp: lookup prometheus.d8-monitoring on 169.254.169.254:53: no such host","operator.component":"taskRunner","queue":"main","task.id":"e629649a-3b5d-4bb1-9f87-0469c6efeb81","time":"2022-05-14T08:42:25Z"}
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'e2e/run/aws'...

**Review by nabokihms — APPROVED** [review]

**Pull request merged**